service/collector/aws.go

```diff
@@ -40,27 +40,52 @@ func (c Collector) getARNs() ([]string, error) {
 // getAWSClients return a list of aws clients for every guest cluster account plus
 // the host cluster account.
 func (c Collector) getAWSClients() ([]aws.Clients, error) {
-	var clients []aws.Clients
+	var (
+		clients    []aws.Clients
+		clientsMap = make(map[string]aws.Clients)
+	)
 
 	arns, err := c.getARNs()
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	// Host cluster account.
-	clients = append(clients, aws.NewClients(c.awsConfig))
+	awsClients := aws.NewClients(c.awsConfig)
+	c.addClient(awsClients, &clientsMap)
```

Comment:
Can we make `addClient` a closure instead? I mean `addClientFunc := func(...) {...}`. That would be way less confusing for me.

Also errors aren't handled.